pkg/secretutils/secret_manager.go

```diff
@@ -96,6 +103,12 @@ func (sm *SecretManager) claimSecret(secret *corev1.Secret, owner client.Object,
 				needsUpdate = true
 			}
 		}
+
+		if !utils.StringInList(secret.Finalizers, SecretsFinalizer) {
```

Comment:
I think we should pass a specific `addFinalizer` flag to this function to trigger adding the finalizer, and do it only for the BMC creds secret.
Already in this patch we're missing removing the finalizer from `info.host.Spec.PreprovisioningNetworkData`, both here and in the PreprovisioningImage controller. Problems like this will only continue to multiply if adding the finalizer is implicit rather than explicit.

Reply:
So do we want to set finalizers for:
 - all secrets
 - all secrets that have ownerRef
 - only BMC secret that has ownerRef
 - only BMC secret regardless of ownerRef
 
 ?

Reply:
^^^ @zaneb @fmuyassarov @andfasano

Reply:
Only BMC secret, which always has ownerRef.